libwild/src/resolution.rs

```diff
@@ -598,11 +601,38 @@ fn assign_section_ids<'data>(
         for file in &mut group.files {
             if let ResolvedFile::Object(s) = file {
                 output_sections.add_sections(&s.custom_sections, s.sections.as_mut_slice(), args);
+                apply_init_fini_secondaries(
+                    &s.init_fini_sections,
+                    s.sections.as_mut_slice(),
+                    output_sections,
+                );
             }
         }
     }
 }
 
+fn parse_priority_suffix(name: &[u8]) -> Option<u16> {
+    let idx = name.iter().rposition(|&b| b == b'.')?;
+    let suffix = name.get(idx + 1..)?;
+    if suffix.is_empty() || !suffix.iter().all(|b| b.is_ascii_digit()) {
+        return None;
+    }
+    let value = core::str::from_utf8(suffix).ok()?.parse::<u32>().ok()?;
+    Some(u16::try_from(value).unwrap_or(u16::MAX))
+}
+
+fn init_fini_priority(name: &[u8]) -> Option<u16> {
+    if name == secnames::INIT_ARRAY_SECTION_NAME || name == secnames::FINI_ARRAY_SECTION_NAME {
+        return Some(u16::MAX);
+    }
+
+    if name.starts_with(b".init_array.") || name.starts_with(b".fini_array.") {
```

Comment:
I think the parsing code might be simplified slightly if you used `strip_prefix`. e.g.:

```rust
if let Some(rest) = name.strip_prefix(b".init_array.") {
    return parse_priority_suffix(rest);
}
```